lib/guardian/topic_guardian.rb

```diff
@@ -199,4 +199,18 @@ def can_edit_tags?(topic)
 
     false
   end
+
+  def can_perform_action_available_to_group_moderators?(topic)
+    return false if anonymous? || topic.nil?
+    return true if is_staff?
+    return true if @user.has_trust_level?(TrustLevel[4])
```

Comment:
Could trust level 4 previously close and archive topics? I didn't think they could!

Reply:
I _think_ so. Please double check the following for me.

canManageTopic is true for isStaff or isElder, where isElder is someone with a trust level of 4:

https://github.com/discourse/discourse/blob/33554e5cbc58463190a8257eb3bd8875b4c3479f/app/assets/javascripts/discourse/app/models/user.js#L218

Having canManageTopic exposes many admin options to someone with a trust level of 4:

https://github.com/discourse/discourse/blob/0d6b638dac161980ee39631271fd5b3f00fe9b14/app/assets/javascripts/discourse/app/widgets/topic-admin-menu.js#L133

Many of these actions (ie, pinned pinned_globally visible closed archived) end up hitting TopicsController#status, with the work happening around lines 413-416:

https://github.com/discourse/discourse/blob/0d6b638dac161980ee39631271fd5b3f00fe9b14/app/controllers/topics_controller.rb#L413-L416

which ends up calling the following:

https://github.com/discourse/discourse/blob/33554e5cbc58463190a8257eb3bd8875b4c3479f/lib/guardian.rb#L163-L165

where trust level 4 is checked against.

Reply:
Well that is very convincing evidence! Thanks for the leg work.